architecture/evm/error_normalizer.go

```diff
@@ -307,14 +355,19 @@ func ExtractJsonRpcError(r *http.Response, nr *common.NormalizedResponse, jr *co
 					nil,
 					details,
 				),
+				true, // retryable
 			)
-		} else if code == -32600 {
+		}
+
+		//----------------------------------------------------------------
+		// 12) "Invalid Request" errors
+		//----------------------------------------------------------------
+
+		if code == -32600 {
 			if dt, ok := err.Data.(map[string]interface{}); ok {
-				if msg, ok := dt["message"]; ok {
-					if strings.Contains(msg.(string), "validation errors in batch") {
-						// Intentionally return a server-side error for failed requests in a batch
-						// so they are retried in a different batch.
-						// TODO Should we split a batch instead on json-rpc client level?
+				if innerMsg, ok := dt["message"]; ok {
+					if strings.Contains(innerMsg.(string), "validation errors in batch") {
+						// Return a server-side error so the caller might retry or split the batch.
```

Comment:
Since we have retryable client errors I think we can remove this edge-case? There's a unit test for that I expect that should pass

Reply:
I think the edge-case itself. cannot be removed cause it seems a specific one that carries a certain error in `data.message`. but I created a branch for it in our now organized `Invalid Request` category which would return a client-side exception which is retryable.